github/state/Downloader.kt

```diff
@@ -0,0 +1,87 @@
+package github.state
+
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import org.kohsuke.github.*
+import java.io.IOError
+import java.nio.file.Paths
+
+class Downloader {
+    enum class State {
+        NOT_STARTED,
+        GITHUB_DOWNLOADING,
+        WRITING_TO_FILE,
+        COMPLETED,
+    }
+    // Need to choose a sane multi-platform place for these to go, can't go in bazel's temporary execution environment.
```

Comment:
I think storing them in the execution environment is fine. The user has no use for the files after closing the application, and it doesn't take long to re-download them if they want to try the same repo another time.

Reply:
I removed this comment, this was for when I was pointing to my own home directory.